Documentation/signing-and-verification-guide.md

```diff
@@ -0,0 +1,265 @@
+
+```
+$ gpg2 --no-default-keyring --secret-keyring ./rocket.sec --keyring ./rocket.pub --list-keys
+```
+```
+./rocket.pub
+------------
+pub   2048R/26EF7A14 2015-01-09
+uid       [ unknown] Kelsey Hightower (ACI signing key) <kelsey.hightower@coreos.com>
+sub   2048R/B9C074CD 2015-01-09
+```
+
+#### Export the public key
+
+```
+$ gpg2 --no-default-keyring --armor \
+--secret-keyring ./rocket.sec --keyring ./rocket.pub \
+--output pubkeys.gpg \
+--export "<kelsey.hightower@coreos.com>"
+```
+
+### Sign the image
+
+```
+$ gpg2 --no-default-keyring --armor \
+--secret-keyring ./rocket.sec --keyring ./rocket.pub \
+--output hello-0.0.1-linux-amd64.sig \
+--detach-sig hello-0.0.1-linux-amd64.aci
+```
+
+#### Verify the image using gpg2
+
+```
+$ gpg2 --no-default-keyring \
+--secret-keyring ./rocket.sec --keyring ./rocket.pub \
+--verify hello-0.0.1-linux-amd64.sig hello-0.0.1-linux-amd64.aci
+```
+```
+gpg: Signature made Fri Jan  9 05:01:49 2015 PST using RSA key ID 26EF7A14
+gpg: Good signature from "Kelsey Hightower (ACI signing key) <kelsey.hightower@coreos.com>" [unknown]
+gpg: WARNING: This key is not certified with a trusted signature!
+gpg:          There is no indication that the signature belongs to the owner.
+Primary key fingerprint: B346 E31D E7E3 C6F9 D1D4  603F 4DFB 61BF 26EF 7A14
+```
+
+At this point you should have the following three files:
+
+* hello-0.0.1-linux-amd64.sig
+* hello-0.0.1-linux-amd64.aci
+* pubkeys.gpg
+
+## Distributing Images via Meta Discovery
+
+Host an HTML page with the following meta tags:
+
+```
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta name="ac-discovery" content="example.com/hello https://example.com/images/{name}-{version}-{os}-{arch}.{ext}">
+    <meta name="ac-discovery-pubkeys" content="example.com/hello https://example.com/pubkeys.gpg">
+  <head>
+<html>
+```
+
+Serve the following files at the locations described in the meta tags:
+
+* hello-0.0.1-linux-amd64.sig
+* hello-0.0.1-linux-amd64.aci
+* pubkeys.gpg
+
+### Rocket Integration
+
+Lets walk through the steps rocket takes when installing images using Meta Discovery.
+The following rocket command:
+
+```
+$ rkt run example.com/hello:0.0.1
+```
+
+translates into the following curl commands:
+
+```
+$ curl https://example.com/hello?ac-discovery=1
+$ curl https://example.com/images/example.com/hello-0.0.1-linux-amd64.aci
+$ curl https://example.com/images/example.com/hello-0.0.1-linux-amd64.sig
+```
+
+The first response contains the template URL used to download the ACI image and detached signature file.
+
+```
+<meta name="ac-discovery" content="example.com/hello https://example.com/images/{name}-{version}-{os}-{arch}.{ext}">
+```
+
+Rocket populates the `{os}` and `{arch}` based on the current running system.
+The `{version}` will be taken from the tag given on the command line.
+The `{ext}` is based on convention and .aci will be used for ACI images and .sig will be used for detached signatures.
```

Comment:
"based on convention" is awkward; perhaps just "the ext will be substituted appropriately depending on the type of thing being retrieved: ".aci" will be used for ..."